WCFDataService/Client/System/Data/Services/Client/ClientEdmModel.cs

```diff
@@ -408,7 +408,14 @@ private EdmTypeCacheValue GetOrCreateEdmTypeInternal(IEdmStructuredType edmBaseT
                             List<IEdmStructuralProperty> loadedKeyProperties = new List<IEdmStructuralProperty>();
                             foreach (PropertyInfo property in ClientTypeUtil.GetPropertiesOnType(type, /*declaredOnly*/edmBaseType != null).OrderBy(p => p.Name))
                             {
-                                IEdmProperty edmProperty = this.CreateEdmProperty((EdmStructuredType)entityType, property);
+                                // A better approach for getting to the value of a property without triggering lazy construction
+                                // could be desired, but I don't control the auto proxy generation stuff, so for now...
+                                // we just favor convwention over configuration and expect to find a field with the same name
+                                // as the property, but prefixed with __ or _.
+                                FieldInfo backingField = type.GetField("__" + property.Name, BindingFlags.NonPublic | BindingFlags.Instance);
```

Comment:
1. Yes. I will refactor the shared bits.
2. No. Only fields (not properties) with the same name and casing as the property, but prefixed with "__" or "_" - will be considered as being the backing field for the property. The standard add service reference and connected service proxy generation utilities generates backing fields prefixed with "_". The addition of "__" is to make it easy to change an autogenerated proxy into one that uses lazy initialization with a RegEx expression. The property names are unchanged, the getter of the properties would trigger lazy initialization.